pkg/controllers/provisioning/controller.go

```diff
@@ -40,11 +40,10 @@ type Controller struct {
 }
 
 // NewController constructs a controller instance
-func NewController(kubeClient client.Client, provisioner *Provisioner, recorder events.Recorder) corecontroller.Controller {
+func NewController(kubeClient client.Client, provisioner *Provisioner) corecontroller.Controller {
 	return corecontroller.Typed[*v1.Pod](kubeClient, &Controller{
 		kubeClient:  kubeClient,
 		provisioner: provisioner,
-		recorder:    recorder,
```

Comment:
I'm imagining that this recorder can just be dropped outright without any other changes to this file